rcl_lifecycle/test/test_default_state_machine.cpp

```diff
@@ -833,3 +834,21 @@ TEST_F(TestDefaultStateMachine, default_sequence_error_unresolved) {
       rcl_lifecycle_state_machine_fini(&state_machine, this->node_ptr, this->allocator));
   }
 }
+
+TEST_F(TestDefaultStateMachine, init_fini_maybe_fail) {
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    // If this in zero initialized outside of this loop, there is potential for a segfault
+    // when attempting to init a second time
```

Comment:
@brawner hmm, shouldn't `fini` take care of leaving `rcl_lifecycle_state_machine_t` in a known state?

Reply:
Looks like this was the case before I finalized a second time. This comment is no longer accurate. Moving zero_initialization out of this test.